docs/source/user-guide/dataset_object.rst

```diff
@@ -0,0 +1,128 @@
+      - float
+      - Used to infer which columns are categorical (if `cat_features` isn't explicitly passed).
+        Set maximum ratio of unique values in a column in order for it to be categorical.
+        The higher the value, the chance of column inferred as categorical is higher
+      - 0.01
+    * - max_categories
+      - int
+      - Used to infer which columns are categorical (if `cat_features` isn't explicitly passed).
+        Set maximum number of unique values in a column in order for it to be categorical.
+        The higher the value, the chance of column inferred as categorical is higher
+      - 30
+    * - max_float_categories
+      - int
+      - Same as `max_categories` but for columns of type float
+      - 5
+    * - convert_date
+      - bool
+      - Whether to convert date column if it's numeric to date
+      - True
+
+Inferring Features And Categorical Features
+================================================
+Dataset defines which columns of the data are features and of them which are categorical features.
+For features, if parameter `features` not passed explicitly, all will be considered features apart from the label, index and date.
+For categorical features, if parameter `cat_features` not passed explicitly, the following logic runs on every column to determine
+whether the column is categorical or not:
+
+* if columns is float type:
+    * number of unique < `max_float_categories`
+* else:
+    *  number of unique < `max_categories` AND (number of unique / number of samples) < `max_categorical_ratio`
+
+Creating a Dataset
+======================
+
+From a DataFrame
+~~~~~~~~~~~~~~~~
+The default dataset constructor is expecting to get a dataframe. the rest of the properties
+are optional, but if your data have date/index/label you would want to define them.
+
+.. code-block:: python
+
+    Dataset(my_dataframe, features=['feat1', 'feat2', 'feat3'], label='target', index='id', date='timestamp')
+
+From Numpy Arrays
+~~~~~~~~~~~~~~~~~~~
+A Dataset can be created using a 2D numpy array for features and 1D numpy array for the labels. The features array is mandatory, and the labels array is optional.
+
+.. code-block:: python
+
+    features = np.array([[0.25, 0.3, 0.3], [0.14, 0.75, 0.3], [0.23, 0.39, 0.1]])
+    labels = np.array([0.1, 0.1, 0.7])
+    dataset_with_labels = Dataset.from_numpy(features, labels)
+    dataset_without_labels = Dataset.from_numpy(features)
+
+Also, it's possible to assign names to the features and label:
+
+.. code-block:: python
+
+    Dataset.from_numpy(
+        features, labels,
+        feature_names=['feat1', 'feat2', 'feat3',],
+        label_name='target'
+    )
+
+All the rest of the Dataset's properties can be passed also as a regular keyword arguments:
+
+.. code-block:: python
+
+    Dataset.from_numpy(
+        features, labels,
+        feature_names=['feat1', 'feat2', 'feat3',],
+        label_name='target',
+        max_float_categories=10
+    )
```

Comment:
I would add also how to copy the dataset configuration from another dataset but with new data (using the copy function)

Reply:
The copy is meant to be used mostly internally by us

Reply:
I agree that Bresslers' example might be useful use-case
let's add this also to the "updating the dataset object doc" task @matanper ?